packages/context/src/resolver.ts

```diff
@@ -136,6 +136,77 @@ function resolve<T>(
   return resolved;
 }
 
+/**
+ * Resolve entries of an object into a new object with the same keys or a
+ * promise of the new object. Please note some of the entries can be resolved
+ * synchronously while others need to be resolved asynchronously.
+ * @param map The original object containing the source entries
+ * @param resolver A function resolves an entry to a value or promise
+ */
+export function resolveMap<T, V>(
+  map: {[name: string]: T},
+  resolver: (v: T, map: {[name: string]: V}, p: string) => ValueOrPromise<V>,
```

Comment:
AFAICT, the second argument `map` (more accurately `result`) is not used anywhere. Could you please remove it? I am concerned that by allowing direct access to the `result` array, users of this methods can easily shoot their own foot.

Ideally, I think the `resolver` signature should be the same as the signature of the callback for [`Array.prototype.map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map).

```js
var new_array = arr.map(function callback(currentValue[, index[, array]]) {
    // Return element for new_array
}[, thisArg])
```

The advantage of passing the original map/list to the callback is avoiding an unnecessary closure if the callback needs to access the original map/list. I would not bother with `thisArg` here.